discovery/kubernetes/endpoints.go

```diff
@@ -411,7 +411,8 @@ func (e *Endpoints) buildEndpoints(eps *apiv1.Endpoints) *targetgroup.Group {
 			continue
 		}
 
-		for _, c := range pe.pod.Spec.Containers {
+		containers := append(pe.pod.Spec.Containers, pe.pod.Spec.InitContainers...)
```

Comment:
- maybe also consider initcontainers in line `364` in case a sidecar has a port exposed in the Endpoints. (it'd be great to have a test for that as well)
- We'll need a label as we have for the Pod role to differentiate them from containers: `__meta_kubernetes_pod_container_init`
- we'll need to do the same in `endpointslice.go`